modules/core/src/main/scala/fs2/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -483,6 +489,7 @@ private[kafka] object KafkaConsumerActor {
   type StreamId = Int
 
   final case class State[F[_], K, V](
+    assignments: Set[TopicPartition],
```

Comment:
I did it this way because I experienced deadlock when I tried to do so with `withConsumer(_.assignments().toSet)`